x-pack/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -527,7 +549,18 @@ export function XYChart({
             case 'bar_stacked':
             case 'bar_horizontal':
             case 'bar_horizontal_stacked':
-              return <BarSeries key={index} {...seriesProps} />;
+              const valueLabelsSettings = {
+                displayValueSettings: {
+                  // TODO: workaround for elastic-charts issue with horizontal bar chart and isValueContainedInElement flag
+                  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                  valueFormatter: (d: any): string => xAxisFormatter.convert(d),
```

Comment:
You're using the wrong formatter here- the `valueFormatter` is the Y axis, not the X axis.